crates/storage/trie.rs

```diff
@@ -24,147 +35,123 @@ pub type ValueRLP = Vec<u8>;
 /// Libmdx-based Ethereum Compatible Merkle Patricia Trie
 /// Adapted from https://github.com/lambdaclass/merkle_patricia_tree
 pub struct Trie {
-    /// Reference to the current root node
-    root_ref: NodeRef,
-    /// Contains the trie's nodes & old root hashes
+    /// Hash of the current node
+    root: Option<NodeHash>,
+    /// Contains the trie's nodes
     pub(crate) db: TrieDB,
-    // Contains the root hash if the current root node has been hashed
-    hash: Option<H256>,
 }
 
 impl Trie {
-    /// Creates a new Trie based on either a previous execution's DB (if `trie_dir` contains a DB) or a clean DB
+    /// Creates a new Trie from a clean DB
     pub fn new(trie_dir: &str) -> Result<Self, StoreError> {
-        let (db, root_ref) = TrieDB::init(trie_dir)?;
         Ok(Self {
-            root_ref: root_ref.unwrap_or_default(),
-            db,
-            hash: None,
+            db: TrieDB::create(trie_dir)?,
+            root: None,
```

Comment:
Shouldn't we put `EMPTY_TRIE_HASH ` to no have to deal with a Option every time?

Reply:
I find it more intuitive to deal with an option than having to check root != EMPTY_TRIE_HASH before any trie operation, it is also easier to avoid errors due to skipping this check in later refactors